cargo-dist/src/init.rs

```diff
@@ -84,15 +128,59 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
         }
     };
 
+    // If we're migrating, the configuration will be missing the
+    // generic workspace specification, and will have some
+    // extraneous cargo-specific stuff that we don't want.
+    if is_migrating {
+        // Remove Cargo-specific workspace
+        workspace_toml.remove("workspace");
+
+        // Write generic workspace config
+        let mut table = toml_edit::table();
+        if let Some(t) = table.as_table_mut() {
+            let mut array = toml_edit::Array::new();
+            array.push("cargo:.");
+            t["members"] = toml_edit::value(array);
+            t.set_position(0);
+        }
+        workspace_toml.insert("workspace", table);
+
+        // These fields will continue to be read from Cargo.toml
+        workspace_toml.remove("dependencies");
+        workspace_toml.remove("dev-dependencies");
+        workspace_toml.remove("package");
+        workspace_toml.remove("profile");
+    }
+
     if let Some(meta) = &multi_meta.workspace {
-        apply_dist_to_workspace_toml(&mut workspace_toml, root_workspace.kind, meta);
+        apply_dist_to_workspace_toml(&mut workspace_toml, desired_workspace_kind, meta);
     }
 
     eprintln!();
 
+    let filename = match desired_workspace_kind {
+        WorkspaceKind::Rust => "Cargo.toml",
+        WorkspaceKind::Generic => "dist-workspace.toml",
+        WorkspaceKind::Javascript => todo!(),
```

Comment:
Also dist-workspace.toml, technically... Javascript is in a weird place in the code -- ends up being defacto generic at the top-level. Generally only shows up internally as a nested workspace (not your problem, just thinking out loud).

Reply:
I was wondering about that, since we have the package.json-only config stuff. I'll add it to the Generic branch.